keps/sig-node/4815-dra-partitionable-devices/README.md

```diff
@@ -0,0 +1,3411 @@
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+Proposed changes:
+```
+// ResourceSliceSpec contains the information published by the driver in one ResourceSlice.
+type ResourceSliceSpec struct {
+		...
+
+		// DeviceMixins represents a list of device mixins, i.e. a collection of
+		// shared attributes and capacities that an actual device can "include"
+		// to extend the set of attributes and capacities it already defines.
+		//
+		// The main purposes of these mixins is to reduce the memory footprint
+		// of devices since they can reference the mixins provided here rather
+		// than duplicate them.
+		//
+		// Must not have more than 128 entries.
+		//
+		// +optional
+		// +listType=atomic
+		DeviceMixins []DeviceMixin
+}
+
+// DeviceMixin defines a specific device mixin for each device type.
+// Besides the name, exactly one field must be set.
+type DeviceMixin struct {
+		// Name is a unique identifier among all mixins managed by the driver
+		// in the pool. It must be a DNS label.
```

Comment:
*in the pool* - I like that, in that it means we only need to define the mixins once per pool. But if someone *does* define them more than once in the pool (in two different slices in the same pool), and they are *not the same*, what do we do? How do we pick the "right" one?

Reply:
We would error out when parsing / consolidating the slices in the scheduler if there is a conflict. It's the same same we already have to do for unique device names across slices.